subprojects/build-init/src/test/groovy/org/gradle/buildinit/plugins/internal/maven/MavenProjectsCreatorSpec.groovy

```diff
@@ -113,4 +113,31 @@ class MavenProjectsCreatorSpec extends Specification {
         def ex = thrown(MavenConversionException)
         ex.message == "Unable to create Maven project model. The POM file $pom does not exist."
     }
+
+    def "creates projects with compileOnly library"() {
```

Comment:
Should probably be renamed to something more expressive like "can translate dependency assigned to Maven provided scope into compileOnly".